airflow/api_connexion/endpoints/dag_warning_endpoint.py

```diff
@@ -57,7 +60,12 @@ def get_dag_warnings(
     allowed_filter_attrs = ["dag_id", "warning_type", "message", "timestamp"]
     query = select(DagWarningModel)
     if dag_id:
+        if not get_airflow_app().appbuilder.sm.can_read_dag(dag_id, g.user):
```

Comment:
I feel we should have the DAG permission in the decorator instead of handling this by ourselves. My vote would go to just adding `(permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG)` to the list of permissions. 
That should solve this. Assuming it's dag_ids instead of dag_id, then we can do as you're doing now.

Reply:
Oh, I see that if dag_id is not provided, then it gets all dag warning, my bad.